tutorials/single_node_tutorial/single_gpu_tutorial.ipynb

```diff
@@ -0,0 +1,3423 @@
+    "cleaned_data = cleaner(lang_data)\n",
+    "\n",
+    "# Write the cleaned_data\n",
+    "write_to_disk(cleaned_data.df, lang_sep_cleaned_data_output_path, write_to_filename=True, output_type='jsonl')\n",
+    "\n",
+    "print(f\"Time taken for fixing unicode:{time.time()-t0}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00c6e5a1",
+   "metadata": {},
+   "source": [
+    "Verify the result. We can see that some documents has been removed from TH wikipedia dataset since the number of lines in this output file is less than the original file (no. of lines = 162164)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "b2b34d46",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "thwiki-20240201-pages-articles-multistream.xml.bz2.jsonl\n",
+      "161748 /nluo_data/NeMo-Curator/tutorials/single_node_tutorial/workspace/language_sep/data/cleaned/thwiki-20240201-pages-articles-multistream.xml.bz2.jsonl\n"
+     ]
+    }
+   ],
+   "source": [
+    "! ls {lang_sep_cleaned_data_output_path}\n",
+    "! wc -l  {lang_sep_cleaned_data_output_path}/thwiki-20240201-pages-articles-multistream.xml.bz2.jsonl"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39d539a2",
+   "metadata": {},
+   "source": [
+    "Furthur verify by loading documents that has been identified as other language, such as 'EN'. We can see from output that the removed document is indeed in English and contains very little or even no Thai."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5ace3c5b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_jsonl_file(os.path.join(language_separated_output_path,'EN'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9b817bf7",
+   "metadata": {},
+   "source": [
+    "**[Optional]**Close the Dask cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 153,
+   "id": "bf05b6c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# client.cluster.close()\n",
+    "# client.shutdown()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cc8b6aef",
+   "metadata": {},
+   "source": [
+    "## 3.Add ID\n",
+    "TH wikipedia data do have `id` field, but the `id` field contains number only. It will be better if we unified the `id` field and transform it to the format of `<prefix>_<id>`. In this way, when handling multiple dataset, we will able to know which document from which dataset has been removed. This `id` will be useful when we are running deduplication and heuristic filtering. The function we will be using is `AddID()`. Arguments for this function include:\n",
+    "- `id_field`: fields will be added to input .json file. If the key already exists in the .jsonl, it's value will be replaced.\n",
+    "- `id_prefix`: prefix used in ID. Default is 'doc-id'\n",
+    "- `start_index`: starting index in ID. Default is 0"
```

Comment:
This recently changed. The default is now `None` and the id is considered "unordered" by default to improve speed.

Reply:
Fixed the description. In the code section, I keep the start_index = 0 for easier reference